apt/minimization/minimizer.py

```diff
@@ -409,40 +486,114 @@ def transform(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, features_names:
+            feature_ncp = self._calc_ncp_categorical(categories[feature], category_counts[feature],
+                                                     self._feature_data[feature],
+                                                     total_count)
+            total_ncp = total_ncp + feature_ncp
+            total_features += 1
+        if total_features == 0:
+            return 0
+        return total_ncp / total_features
+
+    @staticmethod
+    def _calc_ncp_categorical(categories, category_count, feature_data, total):
+        category_sizes = [len(g) if len(g) > 1 else 0 for g in categories]
+        normalized_category_sizes = [s * n / total for s, n in zip(category_sizes, category_count)]
+        average_group_size = sum(normalized_category_sizes) / len(normalized_category_sizes)
+        return average_group_size / feature_data['range']  # number of values in category
+
+    @staticmethod
+    def _calc_ncp_numeric(feature_range, range_count, feature_data, total):
+        # if there are no ranges, feature is suppressed and iLoss is 1
+        if not feature_range:
+            return 1
+        # range only contains the split values, need to add min and max value of feature
+        # to enable computing sizes of all ranges
+        new_range = [feature_data['min']] + feature_range + [feature_data['max']]
+        range_sizes = [b - a for a, b in zip(new_range[::1], new_range[1::1])]
+        normalized_range_sizes = [s * n / total for s, n in zip(range_sizes, range_count)]
+        average_range_size = sum(normalized_range_sizes) / len(normalized_range_sizes)
+        return average_range_size / (feature_data['max'] - feature_data['min'])
+
+    def _get_feature_data(self, x):
+        feature_data = {}
+        for feature in self._features:
+            if feature not in feature_data.keys():
+                fd = {}
+                values = list(x.loc[:, feature])
+                if feature not in self.categorical_features:
+                    fd['min'] = min(values)
+                    fd['max'] = max(values)
+                    fd['range'] = max(values) - min(values)
+                else:
+                    fd['range'] = len(np.unique(values))
+                feature_data[feature] = fd
+        return feature_data
+
     def _get_record_indexes_for_cell(self, X, cell, mapped):
         indexes = []
         for index, row in X.iterrows():
             if not mapped.item(index) and self._cell_contains(cell, row, index, mapped):
                 indexes.append(index)
         return indexes
 
-    def _cell_contains(self, cell, x, i, mapped):
+    def _get_record_count_for_cell(self, X, cell, mapped):
+        count = 0
+        for index, row in X.iterrows():
+            if not mapped.item(index) and self._cell_contains(cell, row, index, mapped):
+                count += 1
+        return count
+
+    def _cell_contains(self, cell, x, index, mapped):
         for f in self._features:
```

Comment:
Why not use enumerate() here too to get both the feature and the index together?